### PR TITLE
WIP: Feature/loading states

### LINKED ISSUE #25: user sees loading screens whenever there is a long background activity
- [x] a user can see a loading state while waiting for the login response 

- [x] a user can see a loading state while waiting for the signup response 

- [x] a user can see a loading state while waiting for create event response

- [x] a user can see a loading state while waiting for create community response

### PR DESCRIPTION
**Pull request by robo360**
- [x] a user can see a loading state while waiting for the login response 

- [x] a user can see a loading state while waiting for the signup response 

- [x] a user can see a loading state while waiting for create event response

- [x] a user can see a loading state while waiting for create community response

- [x] a user can see a loading state while waiting for picture to be uploaded 

Address #25

### CODE REVIEW

File: app/src/main/java/com/example/game/fragments/CreateCommunityFragment.java [hunk 1]
```diff
@@ -86,21 +91,31 @@ public void onClick(View view) {
...
-                community.setImage(new ParseFile(photoFile));
-                community.setDescription(etDescription.getText().toString());
-                community.saveInBackground(new SaveCallback() {
-                    @Override
-                    public void done(ParseException e) {
-                        if (e != null) {
-                            Log.e(TAG, "Error making an event" + e);
-                        } else {
-                            subscribeToCommunity(community);
+                if(etTitle.getText() == null || photoFile == null){
```
Comment (LynnHX) [inline app/src/main/java/com/example/game/fragments/CreateCommunityFragment.java]: nit:space

File: app/src/main/java/com/example/game/fragments/CreateCommunityFragment.java [hunk 2]
```diff
@@ -86,21 +91,31 @@ public void onClick(View view) {
...
+                    Snackbar.make(btnShare, "Title or photo cannot be empty", BaseTransientBottomBar.LENGTH_SHORT).show();
+                    btnShare.setClickable(true);
+                    progressBar.setVisibility(View.GONE);
+                } else{
```
Comment (LynnHX) [inline app/src/main/java/com/example/game/fragments/CreateCommunityFragment.java]: nit:space

File: app/src/main/java/com/example/game/fragments/CreateEventFragment.java [hunk 3]
```diff
@@ -115,16 +118,18 @@ public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceStat
...
         ImageButton ibClose = binding.ibClose;
         TextView tvDisplayDate = binding.tvDisplayDate;
         TextView tvDisplayTime = binding.tvDisplayTime;
         TextView autoTvCommunity = binding.autoTvCommunity;
-        ivPoster = binding.ivPoster;
         tvAddressDisplay = binding.tvAddressDisplay;
+        progressBar = binding.pgBarCreateFragment;
+        ivPoster = binding.ivPoster;
+        btnShare = binding.btnShare;
+
```
Comment (LynnHX) [inline app/src/main/java/com/example/game/fragments/CreateEventFragment.java:132]: nit:redundant new line

File: app/src/main/java/com/example/game/fragments/SignupFragment.java [hunk 4]
```diff
@@ -77,6 +82,8 @@ public void onClick(View view) {
                         public void done(ParseException e) {
                             if (e != null) {
                                 Log.e(TAG, "Error Signup: " + e);
+                                progressBar.setVisibility(View.INVISIBLE);
```
Comment (LynnHX) [inline app/src/main/java/com/example/game/fragments/SignupFragment.java:85]: Is this on purpose? invisible will still take certain part of the the host view

**Review by LynnHX — APPROVED** [review]
Really good structure of stack diff this time! Keep it up! 👍

File: app/src/main/java/com/example/game/fragments/SignupFragment.java (hunk 4 above)
Comment (robo360) [reply app/src/main/java/com/example/game/fragments/SignupFragment.java:85]: yes, it is part of the design. I still want that space.

**Pull request merged**